pkg/api/server.go

```diff
@@ -0,0 +1,49 @@
+package api
+
+import (
+	"OnboardingExercise/pkg/api/lifecycle"
+	"OnboardingExercise/pkg/api/middlewares"
+	"OnboardingExercise/pkg/api/profile"
+	"OnboardingExercise/pkg/repository/profile"
+	"OnboardingExercise/pkg/service/lifecycle"
+	"OnboardingExercise/pkg/service/profile"
+	"github.com/gin-gonic/gin"
+)
+
+type Server struct {
+	engine *gin.Engine
+}
+
+func NewServer() Server {
+	engine := gin.New()
+	server := Server{engine}
+
+	server.initMiddlewares()
+	server.initRoutes()
+
+	return server
+}
+
+func (server Server) Start() error {
+	return server.engine.Run("localhost:8080")
```

Comment:
inject this value to the server from main

Reply:
done